addons/mail/static/src/discuss/core/common/message_model_patch.js

```diff
@@ -1,9 +1,62 @@
 import { Message } from "@mail/core/common/message_model";
+import { Record } from "@mail/core/common/record";
+
 import { patch } from "@web/core/utils/patch";
 
-patch(Message.prototype, {
-    /** @type {Promise[]} */
-    mentionedChannelPromises: [],
+/** @type {import("models").Message} */
+const messagePatch = {
+    setup() {
+        super.setup();
+        this.hasEveryoneSeen = Record.attr(false, {
+            /** @this {import("models").Message} */
+            compute() {
+                return this.thread?.membersThatCanSeen.every((m) => m.hasSeen(this));
+            },
+        });
+        this.hasNewMessageSeparator = Record.attr(false, {
+            compute() {
+                // compute for caching the value and not re-rendering all
+                // messages when new_message_separator changes
+                return this.thread?.selfMember.new_message_separator === this.id;
+            },
+        });
+        this.hasSomeoneFetched = Record.attr(false, {
+            /** @this {import("models").Message} */
+            compute() {
+                if (!this.thread) {
+                    return false;
+                }
+                const otherFetched = this.thread.channel_member_ids.filter(
+                    (m) => m.persona.notEq(this.author) && m.fetched_message_id?.id >= this.id
+                );
+                return otherFetched.length > 0;
```

Comment:
Couldn't we simplify this while we're at it?
```suggestion
                return this.thread?.channel_member_ids.some(
                    (m) => m.persona.notEq(this.author) && m.fetched_message_id?.id >= this.id
                );
```